source/adios2/toolkit/format/bp3/BP3Deserializer.tcc

```diff
@@ -160,9 +160,9 @@ BP3Deserializer::DefineVariableInIO(const ElementIndexHeader &header, IO &io,
     else
     {
         std::lock_guard<std::mutex> lock(m_Mutex);
-        variable =
-            &io.DefineVariable<T>(variableName, characteristics.Shape,
-                                  characteristics.Start, characteristics.Count);
+        Dims zeros(characteristics.Shape.size(), 0);
+        variable = &io.DefineVariable<T>(variableName, characteristics.Shape,
+                                         zeros, characteristics.Shape);
```

Comment:
Let's construct this in the argument, not as neat but that way the mutex portion is explicit to the compiler:
`variable = &io.DefineVariable<T>(variableName, characteristics.Shape, Dims(characteristics.Shape.size(),0), characteristics.Shape);`